src/keysym.c

```diff
@@ -122,6 +128,98 @@ xkb_keysym_is_assigned(xkb_keysym_t ks)
            find_keysym_index(ks) != -1;
 }
 
+struct xkb_keysym_iterator {
+    bool explicit;       /* If true, traverse only explicitly named keysyms */
+    int32_t index;       /* Current index in `keysym_to_name` */
+    xkb_keysym_t keysym; /* Current keysym */
+};
+
+struct xkb_keysym_iterator*
+xkb_keysym_iterator_new(bool explicit)
+{
+    struct xkb_keysym_iterator* iter = calloc(1, sizeof(*iter));
+    iter->explicit = explicit;
+    iter->index = -1;
+    iter->keysym = XKB_KEYSYM_UNICODE_MAX;
+    return iter;
+}
+
+struct xkb_keysym_iterator*
+xkb_keysym_iterator_unref(struct xkb_keysym_iterator *iter)
+{
+    free(iter);
+    return NULL;
+}
+
+xkb_keysym_t
+xkb_keysym_iterator_get_keysym(struct xkb_keysym_iterator *iter)
+{
+    return iter->keysym;
+}
+
+inline bool
+xkb_keysym_iterator_is_explicitly_named(struct xkb_keysym_iterator *iter)
```

Comment:
skip the `inline` here please, it won't be inlined if it's a public function anyway. And `inline` is usually ignored anyway by the compiler, I mostly use it as `static inline` to stop the compiler from complaining about unused functions in headers.